internal/log.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2021 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	logLevelName = map[int64]string{
+		0: "DEBUG",
+		1: "INFO",
+		2: "WARNING",
+		3: "ERROR",
+		4: "CRITICAL",
+	}
+	traceContextRe = regexp.MustCompile(`^(\w+)/(\d+)(?:;o=[01])?$`)
+
+	// maxLogMessage is the largest message that will be logged without chunking, reserving room for prefixes.
+	// See http://cloud/logging/quotas#log-limits
+	maxLogMessage = 255000
+)
+
+func logf(c *context, level int64, format string, args ...interface{}) {
+	if c == nil {
+		panic("not an App Engine context")
+	}
+
+	if !IsSecondGen() {
```

Comment:
The IsSecondGen() function doesn't make sense in the context of a go112+ only version of this library.  maybe use IsAppengine() or IsStandard() here instead?   IsSecondGen() should probably be removed entirely, but maybe not in this PR.

Reply:
Switched to `IsStandard()`